src/BugsnagUnity/Payload/Error.cs

```diff
@@ -137,19 +137,44 @@ internal static Error FromSystemException(System.Exception exception, System.Dia
         {
             var errorClass = exception.GetType().Name;
             var stackFrames = new System.Diagnostics.StackTrace(exception, true).GetFrames();
+           
 
-            StackTraceLine[] lines = null;
-
-            if (stackFrames != null && stackFrames.Length > 0)
+            if (errorClass == ANDROID_JAVA_EXCEPTION_CLASS)
             {
-                lines = new StackTrace(stackFrames).ToArray();
+                string message;
+                var match = Regex.Match(exception.Message, ERROR_CLASS_MESSAGE_PATTERN, RegexOptions.Singleline);
+                // If the message matches the "class: message" pattern, then the Java class is followed
+                // by a description of the Java exception. These two values will be used as the error
+                // class and message.
+                if (match.Success)
+                {
+                    errorClass = match.Groups["errorClass"].Value;
+                    message = match.Groups["message"].Value.Trim();
+                }
+                else
+                {
+                    // There was no Java exception description, so the Java class is the only content in
+                    // the message.
+                    errorClass = match.Groups["errorClass"].Value;
+                    message = "";
+                }
+                var lines = new StackTrace(exception.StackTrace, StackTraceFormat.AndroidJava).ToArray();
+                return new Error(errorClass, message, lines, HandledState.ForUnhandledException(), true);
```

Comment:
perhaps this logic could be contained in something to process class and message, rather than repeating the content of `FromUnityLogMessage()`? for example, both could have something like `(string, string) FormatClassAndMessage(className, rawMessage)`

Reply:
Done 👍  Didint want to adda package dependancy to the sdk so i didnt use a string Tuple, used an array instead

Reply:
Pushed